packages/core/src/lib/base_protocol.ts

```diff
@@ -89,4 +93,15 @@ export class BaseProtocol implements IBaseProtocol {
     // Filter the peers based on the specified criteria
     return filterPeers(allPeersForProtocol, numPeers, maxBootstrapPeers);
   }
+
+  protected ensurePubsubTopicIsValid(
+    pubsubTopic: PubSubTopic,
+    allowedTopics: PubSubTopic[]
+  ): void {
+    if (!allowedTopics.includes(pubsubTopic)) {
+      throw new Error(
+        `Pubsub topic ${pubsubTopic} is not supported by this protocol. Allowed topics are: ${allowedTopics}`
```

Comment:
Again, _allowed_ does not describe it the best.
Same with _supported_. What should a dev do if they encounter the error? probably initialized their node with the correct pubsub topics, right? So we should probably have an exception that tells them that.